ceph/rados/monitor_workflows.py

```diff
@@ -0,0 +1,260 @@
+            )
+            out["unmanaged"] = "true"
+        else:
+            log.debug(
+                f"Setting the service as unmanaged by cephadm. current status : {out}"
+            )
+            out["unmanaged"] = "false"
+
+        file_name = f"/tmp/mon_spec_{self.set_mon_service_managed_type.calls}.yaml"
+        # Creating mon config file
+        self.client.exec_command(sudo=True, cmd=f"touch {file_name}")
+        # Adding the spec rules into the file
+        cmd = f"echo {out} > {file_name}"
+        self.client.exec_command(cmd=cmd, sudo=True)
+
+        log.debug(f"Contents of mon spec file : {out}")
+        apply_cmd = f"ceph orch apply -i {file_name}"
+        log.info(f"Applying the spec file via cmd : {apply_cmd}")
+        self.client.exec_command(cmd=apply_cmd, sudo=True)
+
+        time.sleep(10)
+        # Checking for the unmanaged setting on mon service
+        cmd = "ceph orch ls"
+        out = self.rados_obj.run_ceph_command(cmd=cmd)
+        for entry in out:
+            if entry["service_name"] == "mon":
+                log.debug(f"Service status : {entry}")
+                status = entry.get("unmanaged", False)
+                if status != unmanaged:
+                    log.error(f"Service not in unmamaned={unmanaged} state. Fail")
+                    return False
+                else:
+                    log.info(f"Service  in unmamaned={unmanaged} state. Pass")
+                    return True
+
+    def check_mon_exists_on_host(self, host) -> bool:
+        """
+        Method to check if the mon daemon exists on the given host
+        Args:
+            host: name of the host where existence of mon needs to be checked
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd = f"ceph orch ps {host}"
+        out = self.rados_obj.run_ceph_command(cmd=cmd)
+        for entry in out:
+            if entry["daemon_type"] == "mon":
+                log.debug("Mon daemon present on the host")
+                return True
+        log.debug("Mon daemon not present on the host")
+        return False
+
+    def remove_mon_service(self, host) -> bool:
+        """
+        Method to remove the mon service from the given host
+
+        Args:
+            host: name of the host from where mon needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mon service exists before removing
+
+        if not self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon not present on the host {host}. Returning Pass")
+            return True
+
+        cmd = f"ceph orch daemon rm mon.{host} --force"
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be removed
+        time.sleep(10)
+
+        if self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon still present on the host {host}. Returning Fail")
+            return False
```

Comment:
Shall we change this to a smart wait or is 10 secs enough in every scenario?

Reply:
I have run this multiple times, and it has passed every time.

However I will still go ahead and add a smart check. 

Done!